src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -18,6 +18,10 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
 
   fun findByReference(reference: String): Visit?
 
+  fun findByApplicationReference(applicationReference: String): Visit?
```

Comment:
deleteAllByReferenceIn needs to take an application reference instead.

Reply:
There is another task for this not part of this review